functional_tests/logs/no_drop_logs_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright Splunk Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package nodroplogs
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/signalfx/splunk-otel-collector-chart/functional_tests/internal"
+)
+
+const (
+	testFile           = "test.log"
+	testDir            = "testdata"
+	remoteTestFile     = "/tmp/temp-log-test/test.log"
+	valuesTemplateFile = "no_drop_logs_values.yaml.tmpl"
+	containerName      = "otel-collector"
+	podLabelSelector   = "component=otel-collector-agent"
+)
+
+var podName string
+
+// Env vars to control the test behavior
+// TEARDOWN_BEFORE_SETUP: if set to true, the test will run teardown before setup
+// SKIP_SETUP: if set to true, the test will skip setup
+// SKIP_TEARDOWN: if set to true, the test will skip teardown
+// SKIP_TESTS: if set to true, the test will skip the test
+// UPDATE_EXPECTED_RESULTS: if set to true, the test will update the expected results
+// KUBECONFIG: the path to the kubeconfig file
+func Test_NoDropLogs(t *testing.T) {
+	if os.Getenv("TEARDOWN_BEFORE_SETUP") == "true" {
+		t.Log("Running teardown before setup as TEARDOWN_BEFORE_SETUP is set to true")
+		teardown(t)
+	}
+
+	testKubeConfig, setKubeConfig := os.LookupEnv("KUBECONFIG")
+	require.True(t, setKubeConfig, "the environment variable KUBECONFIG must be set")
+	config, err := clientcmd.BuildConfigFromFlags("", testKubeConfig)
+	require.NoError(t, err)
+	clientset, err := kubernetes.NewForConfig(config)
+	require.NoError(t, err)
+
+	if os.Getenv("SKIP_SETUP") == "true" {
+		t.Log("Skipping setup as SKIP_SETUP is set to true")
+	} else {
+		deployChart(t, testKubeConfig, clientset)
+		deployTestLogToPod(t, clientset, config)
+	}
+
+	if os.Getenv("SKIP_TESTS") == "true" {
+		t.Log("Skipping tests as SKIP_TESTS is set to true")
+		return
+	}
+
+	// 1. Log sink is unavailable, queue becomes full, export fails, and backpressure is applied to the receiver.
+	// This can be verified with the log message "Exporting failed. Rejecting data."
+	// 2. Log sink is available, and all logs are exported successfully.
+	// This can be verified by matching the number of log records to the test log file line count
+	t.Run("NoDropLogs", func(t *testing.T) {
+		time.Sleep(10 * time.Second)
+		podLogs := internal.GetPodLogs(t, clientset, internal.DefaultNamespace, podName, containerName, 100)
+		require.Contains(t, podLogs, "Exporting failed. Rejecting data.", "expected log message not found in pod logs")
+
+		logsConsumer := internal.SetupHECLogsSink(t)
+
+		internal.WaitForLogs(t, 2, logsConsumer)
```

Comment:
Why do we get 2 batches?

Reply:
I am not 100% sure, but I got 2 batches when I run it locally and it matched when I ran it in CI.
I set the sending queue size to 3 , ie: 3 requests, so the filelog receiver must have been sending log in more than 3 requests, resulting of 2 batches.

Reply:
Looks like the filelog receiver batch size is [100](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/3651cff09653d78f72d31380afac73728712d26c/pkg/stanza/fileconsumer/internal/reader/factory.go#L31). We need to figure this out and document to avoid test flakiness

Reply:
> Looks like the filelog receiver batch size is [100](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/3651cff09653d78f72d31380afac73728712d26c/pkg/stanza/fileconsumer/internal/reader/factory.go#L31). It doesn't add up.

Right, this actually makes sense; 
In the test I set the queue size to 3 ( 3 requests ) and as you have shared, the filelog receiver batch size is 100, in other words, the sender queue will be marked full after 3 requests of 100 logs each  ie: 300 logs are consumed with each exporter batch. So we need 2 exporter batch to go through the 600 log test file

Reply:
@dloucasfx do you mind including a comment with the reason for the 2 batches? It'll be helpful context for others who might have to interact with this test